### PR TITLE
add report filter

### PR DESCRIPTION
**Pull request by ghbren**
Added reporter filter support, so we can add something like
```ruby
module Filter
  module TestReportFilter
    def self.filter_report_hash(report_hash)
      report_hash[:config][:builds] ||= {}
      report_hash[:config][:builds][:mykey] = 'myval'
      report_hash
    end
  end

  Salus::Report.register_filter(Filter::TestReportFilter)
end
```

Then the report hash will include `[:config][:builds][:mykey] = 'myval'`

### CODE REVIEW

File: lib/salus/report.rb [hunk 1]
```diff
@@ -40,6 +42,17 @@ def initialize(report_uris: [], builds: {}, project_name: nil, custom_info: nil,
       @ignore_config_id = ignore_config_id # ignore id in salus config
     end
 
+    def apply_report_hash_filters(report_hash)
+      @@filters.each do |filter|
+        report_hash = filter.filter_config(report_hash) if filter.respond_to?(:filter_config)
```
Comment (joshua-ostrom) [inline lib/salus/report.rb:47]: We will want to avoiding overriding filter_config (that hook is for filtering the config).  We should use filter_report_hash here:

```
def apply_report_hash_filters(report_hash)
  @@filters.each do |filter|
    report_hash = filter.filter_report_hash(report_hash) if filter.respond_to?(:filter_report_hash)
  end
  report_hash
end
```
That will ensure that we have different hooks for overriding configs ( filter_config ) and overriding reports hashes ( filter_report_hash )

File: spec/fixtures/blank_repository2/test_plugins/report_filter.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module Filter
+  module TestReportConfig
+    def self.filter_config(report_hash)
```
Comment (joshua-ostrom) [inline spec/fixtures/blank_repository2/test_plugins/report_filter.rb:3]: We should change this to filter_report_hash for good measure.  That will avoid any potential collisions with modules that want to support both hooks (config and report hash)

File: lib/salus/report.rb (hunk 1 above)
Comment (ghbren) [reply lib/salus/report.rb]: Thanks, updated accordingly.

File: spec/fixtures/blank_repository2/test_plugins/report_filter.rb (hunk 2 above)
Comment (ghbren) [reply spec/fixtures/blank_repository2/test_plugins/report_filter.rb]: Thanks, updated accordingly.

**Review by joshua-ostrom — APPROVED** [review]

**Pull request merged**